File: api/src/main/java/org/egglog/api/group/model/dto/response/GroupMemberDto.java
package org.egglog.api.group.model.dto.response;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)

public class GroupMemberDto {
    private Long userId;
    private Long groupId;
    private String userName;
    private String profileImgUrl;
    private Boolean isAdmin;
}

File: client/data/src/main/java/com/org/egglog/data/group/model/GroupMemberDto.kt
package com.org.egglog.data.group.model

import com.org.egglog.domain.group.model.Admin
import com.org.egglog.domain.group.model.GroupMember
import com.org.egglog.domain.group.model.Member
import kotlinx.serialization.Serializable

@Serializable
data class GroupMemberDto(
    val userId : Long?,
    val groupId : Long,
    val userName : String,
    val profileImgUrl : String,
    val isAdmin : Boolean,
    val hospitalName : String
)

fun GroupMemberDto.toDomainModel() : GroupMember {
    return GroupMember(
        userId, groupId, userName, profileImgUrl, isAdmin, hospitalName
    )
}
